lib/db.js

```diff
@@ -467,15 +467,23 @@ module.exports = (
                 // the problem was caused by the unique sessionToken constraint so
                 // return an appropriate error.
                 devices => {
+                  let conflictingDeviceId
+
                   const isDuplicateDeviceId = devices.reduce((is, device) => {
-                    is || device.id === deviceInfo.id
```

Comment:
This looks like it never returned `true`, since arrow functions with braces require the `return` keyword. 😊